### PR TITLE
feat(components): [autocomplete] fetchSuggestions supports Promise

### PR DESCRIPTION
**Pull request by CarterLi**
`<el-autocomplete>` 的 fetchSuggestions 回调函数支持返回promise和直接返回数组

Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [x] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

File: packages/components/autocomplete/src/autocomplete.vue [hunk 1]
```diff
@@ -176,7 +177,13 @@ const getData = (queryString: string) => {
     } else {
       throwError(COMPONENT_NAME, 'autocomplete suggestions must be an array')
     }
-  })
+  }
+  const result = props.fetchSuggestions(queryString, cb)
+  if (isArray(result)) {
+    cb(result)
+  } else if (isPromise(result)) {
+    ;(result as Promise<any[]>).then(cb)
```
Comment (JeremyWuuuuu) [inline packages/components/autocomplete/src/autocomplete.vue:185]: **Question:** Do we really need this cast after the  assertion?

File: packages/components/autocomplete/src/autocomplete.ts [hunk 2]
```diff
@@ -38,7 +38,7 @@ export const autocompleteProps = buildProps({
   },
   fetchSuggestions: {
     type: definePropType<
-      (queryString: string, cb: (data: any[]) => void) => void
+      (queryString: string, cb: (data: { value: string }[]) => void) => void
```
Comment (JeremyWuuuuu) [inline packages/components/autocomplete/src/autocomplete.ts:41]: I think we should properly mark the return type here as in

```ts
type FetchSuggestion = (
(queryString: string,
  cb: (data: { value: string }[]) => void)
) => Array<any> | Promise<Array<any>>
```

**Review by jw-foss — COMMENTED** [review]
I just left a few comments for you to consider, also, would you mind adding a test case against this change? 🏓

**Comment by CarterLi** [comment]
@JeremyWuuuuu Done

**Comment by CarterLi** [comment]
@JeremyWuuuuu Any suggestions?

**Review by jw-foss — APPROVED** [review]

**Comment by JeremyWuuuuu** [comment]
> @JeremyWuuuuu Any suggestions?

LGTM.

**Pull request merged**